src/mdns/mdns_avahi.cpp

```diff
@@ -1024,7 +1024,7 @@ void PublisherAvahi::ServiceSubscription::Resolve(uint32_t           aInterfaceI
     otbrLogInfo("resolve service %s %s inf %d", aInstanceName.c_str(), aType.c_str(), aInterfaceIndex);
     mServiceResolver = avahi_service_resolver_new(
         mPublisherAvahi->mClient, aInterfaceIndex, aProtocol, aInstanceName.c_str(), aType.c_str(),
-        /* domain */ nullptr, AVAHI_PROTO_INET6, static_cast<AvahiLookupFlags>(0), HandleResolveResult, this);
+        /* domain */ nullptr, AVAHI_IF_UNSPEC, static_cast<AvahiLookupFlags>(0), HandleResolveResult, this);
```

Comment:
How does the change work? Do we have any test results?

One small concern about using `AVAHI_IF_UNSPEC` is that if the service has both IPv6 and IPv4 addresses, it may return any of them, which is not ideal because we wanto to prefer IPv6 address.

But it's fine to enhance it in a future PR.